src/modules/oaf/components/OafMask.js

```diff
@@ -0,0 +1,207 @@
+						id="btn-normal-mode"
+						.label=${'Normaler Modus'}
+						.type=${'primary'}
+						@click=${onShowCqlConsole}
+					></ba-button>`
+				: html` <ba-button
+							id="btn-add-filter-group"
+							style="width:200px; display:inline-block; padding: 20px 0px;"
+							.label=${'Neue Filtergruppe'}
+							.type=${'primary'}
+							@click=${onAddFilterGroup}
+						></ba-button>
+						<ba-button
+							id="btn-expert-mode"
+							style="width:200px; display:inline-block;padding: 20px 0px;"
+							.label=${'Expertenmodus'}
+							.type=${'primary'}
+							@click=${onShowCqlConsole}
+						></ba-button>`;
+		};
+
+		const orSeparatorHtml = () => html`
+			<div class="separator-container">
+				<div class="separator"></div>
+				<div class="separator-content">ODER</div>
+				<div class="separator"></div>
+			</div>
+		`;
+
+		const uiModeHtml = () =>
+			html` <div id="filter-groups">
+				${repeat(
+					filterGroups,
+					(group) => group.id,
+					(group, index) => html`
+						<ba-oaf-filter-group
+							group-id=${group.id}
+							@remove=${onRemoveFilterGroup}
+							.queryables=${capabilities.queryables}
+							.oafFilters=${group.oafFilters}
+							@change=${onFilterGroupChanged}
+						></ba-oaf-filter-group>
+						${index < filterGroups.length - 1 ? orSeparatorHtml() : html`<div></div>`}
+					`
+				)}
+			</div>`;
+
+		const consoleModeHtml = () =>
+			html` <div id="console" class="console-flex-container">
+				<div class="btn-bar">${this.operatorDefinitions.map((operator) => html`<ba-button .type=${'primary'} .label=${operator}></ba-button>`)}</div>
+				<textarea class="console"></textarea>
+				<ba-button .type=${'primary'} .label=${'Anwenden'}></ba-button>
+			</div>`;
+
+		return html`
+			<style>
+				${css}
+			</style>
+			<div class="sticky-container">${contentHeaderButtonsHtml()}</div>
+			<div class="container">${showConsole ? consoleModeHtml() : uiModeHtml()}</div>
+		`;
+	}
+
+	_removeFilterGroup(idToRemove) {
+		return this.getModel().filterGroups.filter((group) => {
+			return group.id !== Number(idToRemove);
+		});
+	}
+
+	_findFilterGroupById(id) {
+		return this.getModel().filterGroups.find((group) => {
+			return Number(id) === group.id;
+		});
+	}
+
+	_getLayer() {
+		return this.#storeService
+			.getStore()
+			.getState()
+			.layers.active.find((l) => l.id === this.layerId);
+	}
+
+	async _requestFilterCapabilities() {
+		const layer = this._getLayer();
+		const geoResource = this.#geoResourceService.byId(layer.geoResourceId);
+		const capabilities = await this.#importOafService.getFilterCapabilities(geoResource);
+		this.signal(Update_Capabilities, capabilities);
+	}
+
+	get layerId() {
+		return this.getModel().layerId;
+	}
+
+	set layerId(value) {
+		this.signal(Update_Layer_Id, value);
+	}
+
+	get showConsole() {
+		return this.getModel().showConsole;
+	}
+
+	set showConsole(value) {
+		this.signal(Update_Show_Console, value);
+	}
+
+	get operatorDefinitions() {
```

Comment:
Usually, we would make this a static getter, e.g., `static get OPERATOR_DEFINITIONS`,
and add a test for  its entries